src/passes/OptimizeInstructions.cpp

```diff
@@ -1880,17 +1940,8 @@ struct OptimizeInstructions
       } else {
         // What the reference points to does not depend on the type, so casts
         // may be removable. Do this right before returning because removing a
-        // cast may remove info that we could have used to optimize. For
-        // example:
-        //
-        //  (ref.is_func
-        //    (ref.as_func
-        //      (local.get $anyref)))
-        //
-        // The local has no useful type info. The cast forces it to be a
-        // function, so we can replace the ref.is with 1. But if we removed the
-        // ref.as first then we'd not have the type info we need to optimize
-        // that way.
+        // cast may remove info that we could have used to optimize, see
```

Comment:
Does this apply to `ref.is_null`? Becuase this is within `if (curr->op == RefIsNull)`. The example for this uses `ref.is_func`.

Reply:
Correct, this isn't really an issue for `ref.is_null`. We have limited options there anyhow, and the order could be either way. But for the other code path we must do `skipCast` at the end (new line 1945). We do it the same here for consistency.

Ideally this comment would be on new line 1945... but that's lower down. I could move it and refer to it if that's clearer?

Reply:
Yeah that can be easier to understand. But it would be good to keep "Do this right before returning because removing a cast may remove info that we could have used to optimize" part then.

Reply:
Sounds good, changing.